docs/content/3.middleware/2.guides/11.file-upload.md

```diff
@@ -0,0 +1,200 @@
+        ? parseInt(req.headers["x-upload-size"])
+        : 5242880, // Default to 5MB
+      maxFiles: 5,
+      allowedMimeTypes: ["image/*", "application/pdf"],
+      fieldNames: [],
+    }),
+  }
+);
+```
+
+In this example:
+
+- File uploads are only enabled when the `x-enable-upload: true` header is present.
+- The maximum file size can be controlled via the `x-upload-size` header.
+- Other options remain static but could also be made dynamic based on your needs.
+
+This method is preferred as it provides flexibility and control over file upload behavior, ensuring that uploads are only enabled when necessary and under specific conditions.
+
+## Implementation Example
+
+Here's how to implement a file upload endpoint that connects with the SDK example:
+
+```typescript
+// api/custom-methods/uploadFile.ts
+export async function uploadFile(
+  context: IntegrationContext
+): Promise<CustomMethodResponse> {
+  const files = context.req.files;
+  if (!files) {
+    throw new Error("No files were uploaded");
+  }
+
+  const client = await context.getApiClient("commerce");
+  const formData = new FormData();
+
+  // Handle both multiple files and single file uploads
+  if (Array.isArray(files)) {
+    files.forEach((file) => {
+      formData.append("files", new Blob([file.buffer]), file.originalname);
+    });
+  }
+
+  // Send files to your external service endpoint
+  const response = await client.api.post(
+    "/your-external-service/upload",
+    formData,
+    {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+    }
+  );
+
+  return response.data;
+}
+```
+
+## Usage with SDK
+
+::alert{type="success"}
+For detailed instructions on using file upload with the SDK, visit the [File Upload SDK Guide](https://docs.alokai.com/sdk/advanced/file-upload).
+::
+
+## Custom Configuration
+
+You can customize the file upload behavior by modifying the middleware options:
+
+```typescript
+// middleware.config.js
+module.exports = {
+  integrations: {
+    commerce: {
+      location: "@vue-storefront/commerce-api",
+    },
+  },
+  // Custom file upload configuration
+  fileUpload: {
+    enabled: true,
+    maxFileSize: 10 * 1024 * 1024, // 10MB
+    maxFiles: 10,
+    allowedMimeTypes: ["image/*", "application/pdf", "text/plain"],
+    fieldNames: ["avatar", "documents"], // Specific field names to accept
+  },
+};
+```
+
+::tip
+The middleware automatically handles file size limits, MIME type validation, and file count restrictions based on your configuration.
+::
+
+## Additional Parameters
+
+When uploading files, you can include additional parameters alongside your files. These parameters can be accessed in your middleware implementation:
+
+```typescript
+// api/custom-methods/uploadFile.ts
+export async function uploadFile(
+  context: IntegrationContext
+): Promise<CustomMethodResponse> {
+  const files = context.req.files;
+  const additionalParams = context.req.body; // Access additional parameters
```

Comment:
this won't work?
```suggestion
export async function uploadFile(
  context: IntegrationContext,
  additionalParams: any,
): Promise<CustomMethodResponse> {
  const files = context.req.files;
```

Reply:
Why? This is exactly what I have in my poc and it does work.
```
export async function uploadFile(context: IntegrationContext): Promise<CustomMethodResponse> {
  const files = context.req.files;
  const params = context.req.body;

  console.log("params", params);
  console.log(params.test);
  if (!files) {
    throw new Error("No files were uploaded");
  }

  const formData = new FormData();
  // Handle both multiple files and single file
  if (Array.isArray(files)) {
    files.forEach((file) => {
      // Multer file objects contain buffer and originalname
      formData.append("files", new Blob([file.buffer]), file.originalname);
    });
  }

  return "well done";
}
```

Reply:
Your approach also works and I think it is a better way to do it. I will update the example